### ISSUE TITLE
TestUSAScenario/usa_query_modes fails on ppc64le

### ISSUE THREAD

**Issue by amitsadaphule** [grafana/grafana#39748]
**What happened**:

I was able to build grafana `v8.1.5` and run go unit tests on `RHEL 8.4` ppc64le with the following steps:

```
#!/bin/bash

cd /
PATH=/node-v14.17.6-linux-ppc64le/bin:$PATH
yum install -y wget git && \
    wget https://nodejs.org/dist/v14.17.6/node-v14.17.6-linux-ppc64le.tar.gz && \
    tar -C / -xzf node-v14.17.6-linux-ppc64le.tar.gz && \
    rm -rf node-v14.17.6-linux-ppc64le.tar.gz && \
    npm install -g yarn

cd /
GOPATH=/go
PATH=$PATH:/usr/local/go/bin
yum install -y gcc gcc-c++ wget && \
    wget https://golang.org/dl/go1.17.1.linux-ppc64le.tar.gz && \
    tar -C /usr/local -xzf go1.17.1.linux-ppc64le.tar.gz && \
    rm -rf go1.17.1.linux-ppc64le.tar.gz

mkdir -p $GOPATH/src/github.com/grafana/
cd $GOPATH/src/github.com/grafana/
git clone https://github.com/grafana/grafana.git
cd grafana
git checkout v8.1.5
yarn install --pure-lockfile --no-progress
yarn build
go mod verify
go run build.go setup
go run build.go build
go test -v ./pkg/...
```
However, there was 1 unit test failure, log below:

```
=== RUN   TestUSAScenario/usa_query_modes
2021/09/28 13:10:51 golden file updated: testdata/usa-values-as-rows.txt
    usa_stats_test.go:67: 
        	Error Trace:	usa_stats_test.go:67
        	Error:      	Received unexpected error:
        	            	frame[0] mismatch (-want +got):
        	            	  &data.Frame{
        	            	  	Name: "",
        	            	  	Fields: []*data.Field{
        	            	  		&{Name: "state", vector: &data.stringVector{"AL", "AK", "AZ", "AR", ...}},
        	            	  		&{
        	            	  			Name:   "foo",
        	            	  			Labels: s"",
        	            	  			Config: nil,
        	            	  			vector: &data.float64Vector{
        	            	  				... // 3 identical elements
        	            	  				-0.9791366197808846,
        	            	  				-0.4979007493043646,
        	            	- 				0.6691553056325918,
        	            	+ 				0.6691553056325917,
        	            	  				0.914501128595007,
        	            	  				-0.09980837850296802,
        	            	  				... // 43 identical elements
        	            	  			},
        	            	  		},
        	            	  		&{
        	            	  			Name:   "bar",
        	            	  			Labels: s"",
        	            	  			Config: nil,
        	            	  			vector: &data.float64Vector{
        	            	  				... // 8 identical elements
        	            	  				-0.21488444272885437,
        	            	  				0.8612246693135088,
        	            	- 				0.751062697418599,
        	            	+ 				0.7510626974185991,
        	            	  				-0.39363067135537383,
        	            	  				-0.9961279096447095,
        	            	  				... // 38 identical elements
        	            	  			},
        	            	  		},
        	            	  		&{
        	            	  			Name:   "baz",
        	            	  			Labels: s"",
        	            	  			Config: nil,
        	            	  			vector: &data.float64Vector{
        	            	  				... // 13 identical elements
        	            	  				4.299566258860479,
        	            	  				-0.6063386467095403,
        	            	- 				0.8581051544362406,
        	            	+ 				0.8581051544362407,
        	            	  				-2.4146689575427245,
        	            	  				0.07623121033277871,
        	            	  				... // 33 identical elements
        	            	  			},
        	            	  		},
        	            	  	},
        	            	  	RefID: "",
        	            	  	Meta:  nil,
        	            	  }
        	            	
        	Test:       	TestUSAScenario/usa_query_modes
--- FAIL: TestUSAScenario (0.05s)
    --- FAIL: TestUSAScenario/usa_query_modes (0.05s)
FAIL
FAIL	github.com/grafana/grafana/pkg/tsdb/testdatasource	0.069s
```

**What you expected to happen**:

All unit tests should have passed.

**How to reproduce it (as minimally and precisely as possible)**:

Build+test steps above.

**Anything else we need to know?**:

When I tried the same on amd64, all unit tests passed.

**Environment**:
- Grafana version: v8.1.5
- User OS: RHEL 8.4
- Arch: ppc64le

Any pointers to resolve this issue? Thanks!

**Comment by ryantxu** [comment]
Interesting looks like an issue with the precision saved in the year file.  Perhaps we should clip some digest in the generated data so it is the same on different systems

**Comment by amitsadaphule** [comment]
Thanks for the feedback @ryantxu ! I'll check how this data is generated and raise a PR with changes to clip the digest.

**Comment by amitsadaphule** [comment]
Looks like this issue is already fixed as part of [this](https://github.com/grafana/grafana/commit/377b323fafcbf07ba898cda31086fc95c209cbaa#diff-7e6aaa75855a571e3cc7fe4b492a80f0a5e277095b66a8d4da74d235b21d8b45) commit, but that's not part of **8.1.x** series. Tested the fix by **cherry-pick** in **v8.1.5** tag and the test passed. So, this issue can be closed, I think.
